js/packages/phoenix-client/src/datasets/appendDatasetExamples.ts

```diff
@@ -33,14 +34,15 @@ export async function appendDatasetExamples({
   const inputs = examples.map((example) => example.input);
   const outputs = examples.map((example) => example.output ?? {}); // Treat null as an empty object
   const metadata = examples.map((example) => example.metadata);
+  const datasetInfo = await getDatasetInfo({ client, dataset });
```

Comment:
this incurs an extra network call when dataset name is already available
```suggestion
  let datasetName: string;
  if ("datasetName" in dataset) {
    datasetName = dataset.datasetName;
  } else {
    const datasetInfo = await getDatasetInfo({
      client,
      dataset,
    });
    datasetName = datasetInfo.name;
  }
```

Reply:
that's true, good catch.